beacon_node/network/src/network_beacon_processor/mod.rs

```diff
@@ -842,6 +843,7 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
         block: Arc<SignedBeaconBlock<T::EthSpec, FullPayload<T::EthSpec>>>,
         block_root: Hash256,
         publish_blobs: bool,
+        custody_columns: HashSet<ColumnIndex>,
```

Comment:
Why add this new argument? You can just fetch it at the start of this function
```rust
let is_supernode = self.network_globals.is_supernode();
let custody_columns = self.network_globals.sampling_columns.clone();
```

Reply:
See comments here:
https://github.com/sigp/lighthouse/blob/d6197be93718db58c85c879d5cc3c4971c184a35/beacon_node/network/src/network_beacon_processor/sync_methods.rs#L201-L202

and here
https://github.com/sigp/lighthouse/blob/a5f9649c74aa79abbac90057b0e039c0bc0053eb/beacon_node/network/src/network_beacon_processor/gossip_methods.rs#L1558-L1559

I actually want to take the value from the `GossipVerifiedBlock` and `RpcBlock` but there's a bug in the `RpcBlock` right now, so I've added a `TODO` there.

Reply:
But you have access here to network_globals, what's the need to take it from the block variable? That's only necessary inside the beacon_chain where we can't query network_globals

Reply:
Because DA checker operates on the block `custody_columns_count`, not the global one.
https://github.com/sigp/lighthouse/blob/6f31d44343086ddaa70b396632c0eee724256454/beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs#L187

Network should compute the sampling columns and pass it on to fetch blobs.
If the global sampling columns changes, then this would not match, and block will not get imported.